cmd/vela-email/plugin_test.go

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2021 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+package main
+
+import (
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/jordan-wright/email"
+)
+
+var (
+	mockEmail = &email.Email{
+		To:   []string{"one@mail.com"},
+		From: "two@mail.com",
+	}
+
+	mockSMTPHost = &SMTPHost{
+		Host:     "smtphost.com",
+		Port:     "587",
+		Username: "username",
+		Password: "password",
+	}
+	noAttachment = &email.Attachment{
+		Filename: "",
+	}
+
+	mockBuildEnv = &BuildEnv{
+		BuildCreated:  time.Unix(int64(1556720958), 0).UTC().String(),
+		BuildEnqueued: time.Unix(int64(1556720958), 0).UTC().String(),
+		BuildFinished: time.Unix(int64(1556720958), 0).UTC().String(),
+		BuildStarted:  time.Unix(int64(1556720958), 0).UTC().String(),
+	}
+
+	mockPlugin = &Plugin{
+		Email:      mockEmail,
+		SMTPHost:   mockSMTPHost,
+		Attachment: noAttachment,
+		BuildEnv:   mockBuildEnv,
+	}
+)
+
+func createMockEnv() {
+	os.Setenv("VELA_BUILD_CREATED", "1556720958")
+	os.Setenv("VELA_BUILD_ENQUEUED", "1556720958")
+	os.Setenv("VELA_BUILD_FINISHED", "1556720958")
+	os.Setenv("VELA_BUILD_STARTED", "1556720958")
+	os.Setenv("VELA_BUILD_AUTHOR", "octocat")
+	os.Setenv("VELA_BUILD_AUTHOR_EMAIL", "octocat@github.com")
+	os.Setenv("VELA_BUILD_BRANCH", "main")
+	os.Setenv("VELA_BUILD_COMMIT", "7fd1a60b01f91b314f59955a4e4d4e80d8edf11d")
+	os.Setenv("VELA_BUILD_LINK", "https://vela-server.localhost/octocat/hello-world/1")
+	os.Setenv("VELA_BUILD_MESSAGE", "Merge pull request #6 from octocat/patch-1")
+	os.Setenv("VELA_BUILD_NUMBER", "1")
+	os.Setenv("VELA_REPO_FULL_NAME", "octocat/hello-world")
+}
+
+func TestValidateSuccess(t *testing.T) {
+	tests := []struct {
+		name       string
+		parameters Plugin
+	}{
+		{
+			name: "return no errors: single To email",
+			parameters: Plugin{
+				Email:      mockEmail,
+				SMTPHost:   mockSMTPHost,
+				Attachment: noAttachment,
+			},
+		},
+		{
+			name: "return no errors: multiple To emails",
+			parameters: Plugin{
+				Email: &email.Email{
+					To:   []string{"one@gmail.com", "two@comcast.net"},
```

Comment:
Adding on to this, there are some RFCs around that outline some ways of referencing technical things in documentation and examples. For emails and top level domains and the like, there is this RFC https://datatracker.ietf.org/doc/html/rfc2606 which has couple of options to pick from that should work for tests and docs. Specifically example.com is great because if you navigate to that website it even explicitly says you can use it without permission.